src/bsoncxx/document/view.cpp

```diff
@@ -162,7 +162,9 @@ view::iterator view::find(stdx::string_view key) const {
     bson_t b;
     bson_iter_t iter;
 
-    bson_init_static(&b, _data, _length);
+    if (!bson_init_static(&b, _data, _length)) {
```

Comment:
Only when there's an issue with the length of the document or the _data ptr.  I ran into this as a problem when I was testing, and it might mean that there is a different bug in this class, or that my tests were doing something really wrong.  Maybe returning end() isn't the best way to handle this but we should still probably check this return value.